pangeo_forge_recipes/config.py

```diff
@@ -0,0 +1,35 @@
+from dataclasses import dataclass, field
+from typing import Optional, Union
+
+from pangeo_forge_recipes.storage import CacheFSSpecTarget, FSSpecTarget
+from pangeo_forge_recipes.transforms import RequiredAtRuntimeDefault
+
+
+@dataclass
+class RecipeConfig:
+    """a simple class that SHOULD house all injection spec values
+    from `pangeo_forge_recipes.injections.py:get_injection_specs()`
+
+    This way folks can use it to pass injection spec values around
+    on to custom transforms or partials in their recipe:
+
+    ```python
+    from config import RecipeConfig
+
+    @dataclass
+    class MyCustomTransform(beam.PTransform):
+        # note that `MyCustomTransform` is not part of the
+        # pangeo_forge_recipes.injections.py:get_injection_specs()
+        # but now it can be used anyway in practice
+        target_root: None
+
+    recipe_config = RecipeConfig()
+
+    beam.Create() | MyCustomTransform(target_root=recipe_config.target_root)
+    ```
+    """
+
+    target_root: Union[str, FSSpecTarget, RequiredAtRuntimeDefault] = field(
+        default_factory=RequiredAtRuntimeDefault
+    )
+    cache: Optional[Union[str, CacheFSSpecTarget]] = ""
```

Comment:
while we're talking about naming, can you name this `input_cache` and maybe add the `metadata_cache` field here as well?

Reply:
> `metadata_cache`

will add 

> `input_cache`

as long as folks are comfortable with this but it's not  backward compatible (or haven't thought about how to achieve that yet)

Reply:
MetadataCacheStorage in runner exists for backward compatibility with recipes < 0.10.0. You are correct that it does not exist in recipes >= 0.10.0, and fwiw this is not because InputCacheStorage is playing two roles, but rather because metadata caching is not part of the >= 0.10.0 paradigm.

Reply:
Re: naming, there are (unfortunately) two naming conventions for (roughly) the same things here:

- names in the context of injections
- names in the context of kwargs on PTransforms

I suggest we chose one of the two and just adhere to it exactly, rather than introducing a third convention.

Seems like the injections names are more generic-sounding, and verbose, so that is probably the right one to use, meaning the fields on Config would be:

- Config.target_storage
- Congig.input_cache_storage